Device/Device.cs

```diff
@@ -68,36 +60,44 @@ public async Task ConnectDevice()
             mqttClient.UseDisconnectedHandler(new MqttClientDisconnectedHandlerDelegate(e => Disconnected(e, options)));
         }
 
-        public async Task SendCloudToDeviceMessageAsync(string payload)
-        {
-            var topicD2C = $"devices/{deviceId}/messages/events/";
-
-            Console.WriteLine($"Topic:{topicD2C} Payload:{payload}");
-
+        public MqttApplicationMessage ConstructMessage(string topic, string payload, bool retainFlag = false)
+        {
+            /*var payloadJObject = new JObject
+                {
+                    { "OfficeTemperature", payload + DateTime.UtcNow.Millisecond.ToString() }
+                };
+
+            payload = JsonConvert.SerializeObject(payloadJObject);*/
+            Console.WriteLine($"Topic:{topic} Payload:{payload}");
+
             var message = new MqttApplicationMessageBuilder()
-                .WithTopic(topicD2C)
+                .WithTopic(topic)
                 .WithPayload(payload)
+                .WithRetainFlag(retainFlag)
                 .WithAtLeastOnceQoS()
-                .Build();
+                .Build();
+
+            return message;
+        }
+
+
+        public async Task SendDeviceToCloudMessageAsync(string payload, bool retainFlag = false)
+        {
+            var topicD2C = $"devices/{deviceId}/messages/events/$.ct=application%2Fjson&$.ce=utf-8";

```

Comment:
Move variable topicD2C to ConstructMessage method since it's only used there

Reply:
Not sure about moving `topicD2C` to `ConstructMessage()`. Passing it from outside allows us to construct a message that can be sent to a different topic (even though we're currently not doing it).

Reply:
Then perhaps, we should add it as a parameter for this method as well?

Reply:
@EliiseS I don't disagree with adding `topicD2C` as a param to `SendDeviceToCloudMessageAsync()`. The only caveat would be to then parse `deviceId` from the connection string twice - once inside this class constructor and another in the caller of `SendDeviceToCloudMessageAsync()`. May I suggest we postpone this until we see if other tests actually need to change the topic name at all? If it will always be the same, we could move it inside `ConstructMessage()` like you originally suggested.

Reply:
Sure, sure :)